core/templates/dev/head/components/ck-editor-helpers/ck-editor-4-rte.directive.ts

```diff
@@ -44,13 +44,16 @@ angular.module('oppia').directive('ckEditor4Rte', [
           ContextService.getPageContext() === PAGE_CONTEXT.TOPIC_EDITOR ||
           ContextService.getPageContext() === PAGE_CONTEXT.STORY_EDITOR ||
           ContextService.getPageContext() === PAGE_CONTEXT.SKILL_EDITOR);
+        var contextIsExplorationPlayer = (
+          ContextService.getPageContext() === PAGE_CONTEXT.EXPLORATION_PLAYER);
 
         _RICH_TEXT_COMPONENTS.forEach(function(componentDefn) {
           if (!((scope.uiConfig() &&
             scope.uiConfig().hide_complex_extensions &&
             componentDefn.isComplex) ||
             (!canUseFs && componentDefn.requiresFs) ||
-            (!contextIsLessonRelated && componentDefn.isLessonRelated))) {
+            (!contextIsLessonRelated && componentDefn.isLessonRelated)) ||
+            (contextIsExplorationPlayer)) {
```

Comment:
I think this "if" statement is becoming too complex. Let's try and frame things more generally so that it's more conceptually clear:

- It looks like canUseFs should be renamed canWriteToFs, since we're enabling reading images from the filesystem in the learner view.
- That condition should not be used here, because we always want all the components to exist regardless. It should instead be applied in the place where we're forbidding users from adding new RTE components of that type, or editing existing ones.
- We should also get rid of the contextIsExplorationPlayer clause, since the restriction we're trying to encode is really along the lines of canWriteToFs and isn't actually specific to a particular page.

Also, @aks681 -- should contextIsLessonRelated include the exploration player and editor as well? It seems to me that skillReview components would be used in explorations too?

Reply:
> That condition should not be used here, because we always want all the components to exist regardless. It should instead be applied in the place where we're forbidding users from adding new RTE components of that type, or editing existing ones.

@seanlip , I didn't completely understand what you meant, are you saying we should 
remove ` (!canUseFs && componentDefn.requiresFs)` and `contextIsExplorationPlayer`
from the if statement(removing this will  allow rendering of the required components including images)? If we do that should we still keep the variable `canUseFs` (`canWriteToFs`)?

Reply:
Yes, but keep canWriteToFs since you'll need to refer to that instead, in the separate section where you add the checks for "can add or edit image".

Reply:
As of now, `contextIsLessonRelated` is only available in the topic and skill editors. I think it should only be available in the curated explorations? or all of them?
If the former, it could be done as part of the Android validation PR.

Reply:
Oh yeah, good point. I would say only the curated explorations for now. Let's leave that to a separate PR then. Thanks!